### DISCUSSION TITLE
Generating shelf/bin labels from default_location

### DISCUSSION THREAD

**Discussion by LavissaWoW** [inventree/InvenTree#6052] [Show and tell]
Following the merging of #6016, I was able to put together a very simple plugin (Not yet on GH) that exposes a list of PartBriefs to StockLocation label contexts. 
Effectively making an API call for "part/?default_location=instance.pk" and including the result as context. 

As we have fixed locations for all parts, we label bins with their intended contents.
Combining this and the Brother plugin allowed me to completely integrate the label printing into InvenTree.
(Yes, we actually bought a printer based on the Brother QL plugin :> )

The end result after some template tinkering was this:
![image](https://github.com/inventree/InvenTree/assets/7506688/0f77b4fb-c133-45a3-bf67-ca9d2406f617)

**Comment by SchrodingersGat** [comment]
@LavissaWoW very nice!

**Comment by tnemrap** [comment]
@LavissaWoW  Can you share some code for the label?

**Reply by LavissaWoW** [reply]
Sure. I'll snip the template when I have the chance.
As i've mentioned, the parts listed are based on the plugin I wrote to get the data exposed to the label.
So you can't just add the template to a standard InvenTree installation and have it work. 

Given that heads up, you can expect the template code here within a day or so :)

**Reply by LavissaWoW** [reply]
```html



<style>
	@page {
		width: 188mm;
		height: 38mm;
		padding: 0mm 0mm 1mm 0mm;
		margin: 0px 0px 0px 0px;
 	}
 	
 	.item {
 		width: 100%;
 		height: 100%;
		font-family: Helvetica, Arial, sans-serif;
 	}
 	
 	.desc {
 		width: 100%;
 		background-color: black;
 		color: white;
 		font-weight: bold;
 		font-size: 6mm;
 		text-align: center;
 		padding: 1px;
 	}
 	
 	div.parts {
 		{% if default_parts|length > 4 %}
 		width: 45%;
 		{% else %}
 		width: 100%;
 		{% endif %}
 		height: 100%;
 	}
 	
 	.parts {
 		{% if default_parts|length > 4 %}
	 	flex-basis: 49%;
 		{% else %}
	 	flex-basis: 100%;
 		{% endif %}
 		display: flex;
 		flex-direction: column;
 		align-items: stretch;
		overflow: hidden;
		text-overflow: ellipsis;
 	}
 	
 	.part {
 		flex-grow: 1;
 		flex-basis: 21%;
		white-space: nowrap;
		width: 100%;

 	}
 	
 	.container {
 		display: flex;
 		flex-direction: row;
 		justify-content: space-between;
 	}
 	
</style>

<div class="item">
	<div class="desc">{{ location.description }}</div>
	<div class="container">
		<div class="parts">
			{% for part in default_parts %}
			{% if forloop.counter0 < 4 %}		
			<div class="part">
			<span style="font-weight: bold;font-size: 5.5mm;">{{ part.IPN|default:"No P/N" }}{% if part.revision %}-{{ part.revision }}{% endif %}</span> - {{ part.name }}</br>
			</div>
			{% endif %}		
			{% endfor %}	
		</div>
		{% if default_parts|length > 4 %}
		<div class="parts">
			{% for part in default_parts %}
			{% if forloop.counter0 >= 4 %}		
			<div class="part">
			<span style="font-weight: bold;font-size: 5.5mm;">{{ part.IPN }}{% if part.revision %}-{{ part.revision }}{% endif %}</span> - {{ part.name }}</br>
			</div>
			{% endif %}		
			{% endfor %}	
		</div>
		{% endif %}
	</div>
</div>
```

The `default_parts` parameter isn't exposed via InvenTree core, as mentioned.
There might also be some redundant HTML/CSS in place, as I haven't fine polished the template yet.

**Comment by tnemrap** [comment]
@LavissaWoW are you planing to put the plugin on GH?

**Reply by LavissaWoW** [reply]
https://github.com/LavissaWoW/inventree-label-context

Here it is. 
Do understand that this is a temporary plugin, as I'm planning to add it to the InvenTree core code before 0.14.0. 
But for now, it's usable. 

Oh, and it's not a python package, so you have to git clone it into the plugins directory.
